code/game/machinery/medical_pod/medical_pod.dm

```diff
@@ -0,0 +1,236 @@
+		return
+	if(main_character != M)
+		to_chat(main_character, SPAN_NOTICE("You move [M.name] inside \the [src]."))
+	else
+		to_chat(main_character, SPAN_NOTICE("You move into \the [src]."))
+
+	M.forceMove(src)
+	occupant = M
+	update_use_power(USE_POWER_ACTIVE)
+	update_icon()
+	//prevents occupant's belonging from landing inside the machine
+	for(var/obj/O in src)
+		O.forceMove(loc)
+
+/obj/structure/machinery/medical_pod/proc/go_out()
+	if(!(src.occupant))
+		return
+	for(var/obj/O in src)
+		O.forceMove(loc)
+
+	occupant.forceMove(loc)
+	occupant.update_med_icon()
+
+	if(exit_stun)
+		occupant.stunned = max(occupant.stunned, exit_stun) //Action delay when going out
+		occupant.update_canmove() //Force the delay to go in action immediately
+		occupant.visible_message(SPAN_WARNING("[occupant] pops out of \the [src]!"),
+		SPAN_WARNING("You get out of \the [src] and get your bearings!"))
+
+	occupant = null
+	update_use_power(USE_POWER_IDLE)
+	update_icon()
+	playsound(src, 'sound/machines/hydraulics_3.ogg')
+
+// attackby code
+
+/obj/structure/machinery/medical_pod/attackby(obj/item/W, mob/user)
+
+	if(!ishuman(user))
+		return // no
+	if(inoperable())
+		to_chat(user, SPAN_NOTICE("\The [src] is non-functional!"))
+		return
+	if(istype(W, /obj/item/grab))
+		var/mob/to_put_in
+		var/obj/item/grab/G = W
+		if(istype(G.grabbed_thing,/obj/structure/closet/bodybag/cryobag))
+			var/obj/structure/closet/bodybag/cryobag/C = G.grabbed_thing
+			if(!C.stasis_mob)
+				to_chat(user, SPAN_WARNING("\The [C] is empty!"))
+				return
+			to_put_in = C.stasis_mob
+			C.open()
+			user.start_pulling(to_put_in)
+		else if(ismob(G.grabbed_thing))
+			to_put_in = G.grabbed_thing
+		else
+			return
+		if(skilllock)
+			if(!skillcheck(usr, SKILL_SURGERY, skilllock))
+				to_chat(usr, SPAN_WARNING("You don't have the training to use \the [src]!"))
+				return
+		if(occupant)
+			to_chat(user, SPAN_WARNING("\The [src] is already occupied!"))
+			return
+		if(to_put_in.abiotic())
+			to_chat(user, SPAN_WARNING("Subject cannot have abiotic items on."))
+			return
+		if(isXeno(to_put_in))
+			to_chat(user, SPAN_WARNING("An unsupported lifeform was detected, aborting!"))
+			return
+
+		if(push_in_timer)
+			visible_message(SPAN_NOTICE("[user] starts putting [to_put_in] into \the [src]."), null, null, 3)
+			if(!do_after(usr, push_in_timer, INTERRUPT_NO_NEEDHAND, BUSY_ICON_GENERIC))
+				return
+			if(src.occupant)
+				to_chat(usr, SPAN_NOTICE("\The [src] is already occupied!"))
+				return
+			if(!G || !G.grabbed_thing)
+				return
+
+		go_in(to_put_in, user)
+		add_fingerprint(user)
+
+/obj/structure/machinery/medical_pod/autodoc/attackby(obj/item/W, mob/living/user)
+	. = ..()
+	if(istype(W, /obj/item/stack/sheet/metal))
+		if(stored_metal == max_metal)
+			to_chat(user, SPAN_WARNING("\The [src] is full!"))
+			return
+		var/obj/item/stack/sheet/metal/M = W
+		var/sheets_to_eat = (round((max_metal - stored_metal), 100))/100
```

Comment:
might be a good time to refactor these absurd numbers into something more reasonable to look at

like 5 sheets - 1 limb or whatever the converting factor is

Reply:
erh cba also big number better for future maintainability